src/SAML2/WebSSO/Types.hs

```diff
@@ -25,7 +157,48 @@ import URI.ByteString  -- FUTUREWORK: should saml2-web-sso also use the URI from
 import qualified Data.List as L
 import qualified Data.Text as ST
 import qualified Data.X509 as X509
+import qualified Foundation.Network.IPv4 as IPv4
+import qualified Foundation.Network.IPv6 as IPv6
+import qualified Foundation.Parser as IP
+import qualified Network.DNS.Utils as DNS
 import qualified Servant
+import qualified Text.Email.Validate as Email
+
+
+-- | Text that needs to be escaped when rendered into XML.  See 'mkXmlText', 'mkXmlText'.
```

Comment:
Might also be worth mentioning that `Text` is not allowed to be anywhere in types anymore.